sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/ViewDescriptorFactory.scala

```diff
@@ -255,19 +265,45 @@ private[impl] object ViewDescriptorFactory extends ComponentDescriptorFactory {
       component: Class[_],
       tableType: Class[_],
       tableName: String,
-      tableProtoMessageName: String) = {
+      tableProtoMessageName: String): Seq[KalixMethod] = {
     var previousEntityClass: Option[Class[_]] = None
 
     import ReflectionUtils.methodOrdering
 
-    def invalidComponentException(method: Method, stateClass: Class[_], extraMessage: String) = {
+    def invalidParametersException(method: Method, stateClass: Class[_], extraMessage: String) = {
       throw InvalidComponentException(
         s"Method [${method.getName}] annotated with '@Subscribe.ValueEntity' should either receive " +
         s"a single parameter of type [${stateClass.getName}] or " +
         s"two ordered parameters of type [${tableType.getName}, ${stateClass.getName}]. $extraMessage")
     }
 
-    component.getMethods
+    def invalidParametersForHandleDeletesException(method: Method, extraMessage: String) = {
+      throw InvalidComponentException(
+        s"Method [${method.getName}] annotated with '@Subscribe.ValueEntity.HandleDeletes' should have any parameters. $extraMessage")
+    }
+
+    validateIfHandleDeletesMethodsMatchesSubscriptions(component)
+
+    val handleDeletesMethods = component.getMethods
+      .filter(hasHandleDeletes)
+      .sorted
+      .map { method =>
+        method.getParameterTypes.toList match {
+          case params if params.nonEmpty =>
+            invalidParametersForHandleDeletesException(method, s"Found ${params.size} method parameters.")
+          case _ => // happy days, dev did good with the signature
+        }
+
+        val methodOptionsBuilder = kalix.MethodOptions.newBuilder()
+        methodOptionsBuilder.setEventing(eventingInForValueEntityHandleDeletes(method))
+        addTableOptionsToUpdateMethod(tableName, tableProtoMessageName, methodOptionsBuilder, transform = true)
+
+        KalixMethod(HandleDeletesServiceMethod(method))
+          .withKalixOptions(methodOptionsBuilder.build())
+          .withKalixOptions(buildJWTOptions(method)) //TODO JWT needed?
```

Comment:
Not sure about JWT options.

Reply:
Not entirely sure the return of the method would be hooked up to view updates, probably not, if not, i am not sure what would happen when opening up the ACL and calling from the outside. The view update effect can't become a response automagically.

Reply:
Yeah, personally I wouldn't expose subs methods at all, because it could lead to a very nasty problems with consistency.  At the same time, I know that it might be handy to use them "for a good purpose". Tricky.